### PR TITLE
fix(behavior_path_planner): avoid node death when points of path are empty in some cases

### PR DESCRIPTION
**Pull request by shulanbushangshu**
Signed-off-by: jack.song <jack.song@autocore.ai>

## Description
When  points of path are empty in some cases,the behavior_path_planner will die because of some functions like findEgoSegmentIndex(),findEgoIndex().
The points of path are empty in some cases in the function of getCenterLinePath() in the behavior_path_planner.
The function shall ensure that the forward length is not less than the backward length.
Screenshot from 2023-01-05 12-46-15

## Related links
https://github.com/autowarefoundation/autoware.universe/issues/2607

## Tests performed
Before the pr:
Screenshot from 2023-01-03 15-30-06

After the pr:
Screenshot from 2023-01-05 10-44-41

No matter whether the lane change distance is enough, the node of behavior_path_planner will not die

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/utilities.cpp [hunk 1]
```diff
@@ -1859,6 +1859,8 @@ PathWithLaneId getCenterLinePath(
     s_forward = std::min(s_forward, goal_arc_coordinates.length - lane_change_buffer);
   }
 
+  s_forward = std::max(s_forward, s_backward);
```
Comment (zulfaqar-azmi-t4) [inline planning/behavior_path_planner/src/utilities.cpp:1862]: This makes sense.
I would suggest use if else and put `RCL_WARN` to inform the user that the goal placement for lane change is invalid.

**Comment by zulfaqar-azmi-t4** [comment]
Note: as mentioned https://github.com/autowarefoundation/autoware.universe/issues/2607#issuecomment-1378304322, it will be very helpful to get a map for testing this issue.

**Comment by shulanbushangshu** [comment]
@mitsudome-r 
I will resolve the conflict.

**Comment by shulanbushangshu** [comment]
I test the issue in the autoware of humble.There is still the same problem.I have updated the pr by adding "RCLCPP_WARN".Review required

**Comment by mitsudome-r** [comment]
@zulfaqar-azmi-t4 Could you review the PR?
The map is uploaded in the original issue.

**Comment by shulanbushangshu** [comment]
I test the newest code.It is fixed.
Screenshot from 2023-04-13 11-23-42

**Pull request closed without merging**